cql3/values.hh

```diff
@@ -117,6 +121,56 @@ public:
         return !(*this == other);
     }
 
+    template <typename Func>
+    requires std::invocable<Func, const managed_bytes_view&> && std::invocable<Func, const fragmented_temporary_buffer::view&>
+    decltype(auto) with_value(Func f) const {
+        return f(std::get<fragmented_temporary_buffer::view>(_data));
+    }
+
+    template <typename Func>
+    requires std::invocable<Func, bytes_view>
+    decltype(auto) with_linearized(Func f) const {
+        return with_value([&] (const FragmentedView auto& v) {
+            return ::with_linearized(v, std::forward<Func>(f));
+        });
+    }
+
+    size_t size_bytes() const {
+        return with_value([&] (const FragmentedView auto& v) {
+            return v.size_bytes();
+        });
+    }
+
+    template <typename ValueType>
+    ValueType deserialize(const abstract_type& t) const {
+        return value_cast<ValueType>(with_value([&] (const FragmentedView auto& v) { return t.deserialize(v); }));
+    }
+
```

Comment:
Minor: we call abstract_type::deserialize which arranges for a high-level data_value to be produced, complete with a lw_shared_ptr for the type, then throw the type away. We should have a lower-level entry point that allows us to avoid that useless work. It can be done later.

Reply:
I agree. It a bit insane that there's a memory allocation involved to read an integer from memory.